netlib/http2/__init__.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import (absolute_import, print_function, division)
+import itertools
+import logging
+
+from .frame import *
+from .. import utils
+
+log = logging.getLogger(__name__)
```

Comment:
So, we don't use the built-in logging stuff for output anywhere. If you care about controlling output, the builtin logger is horrible, unless you really contort it to fit. A library imported by a library imported by a library can change the output of your program by changing their logging, which then bubbles up all the way to the "top" of your program. This violates encapsulation and my expectations, and is something that's caused issues for me in practice.

For both mitmdump and pathod/pathoc we use simpler mechanisms, and I would favour avoiding the logging module here too. In fact, I suspect we can remove the logging calls here. If we want to give the user output (say in mitmdump), we need to have the API structured so we can check the result of an operation and craft a message accordingly. If this is really necessary for debugging during dev (it looks like it might not be), there might be better options that don't affect the global application state.